c10/core/thread_pool.cpp

```diff
@@ -1,8 +1,22 @@
 #include <c10/core/thread_pool.h>
 #include <c10/util/Logging.h>
+#include <cpuinfo.h>
 
 namespace c10 {
 
+size_t TaskThreadPoolBase::defaultNumThreads() {
+  cpuinfo_initialize();
+  size_t num_threads = cpuinfo_get_processors_count();
+  if (num_threads > 0) {
+    return num_threads;
+  }
```

Comment:
Does it make more sense to invoke `cpuinfo_get_cores_count` instead? It is preferrable to only utilize physical cores in the case of HT on. `cpuinfo_get_cores_count` makes sure we always utilize all physical CPU cores in both HT on and HT off.

Reply:
It makes sense. However, cpuinfo_get_cores_count returns the same value as cpuinfo_get_processors_count on Linux as you can verify...

Reply:
I'm concerned about the changes of the behavior when HT is on. Previously, only number of physical cores would be used. Now it would be all cores including HT cores, isn't it?